From a4d7dcd55304001db2f70268562d12017aae2527 Mon Sep 17 00:00:00 2001
From: Kendall Willis <k-willis@ti.com>
Date: Tue, 14 Oct 2025 13:40:57 -0500
Subject: [PATCH] TI: Revert "TI: HACK: arm64: dts: ti: Configure am62p
 pinctrl-single for wake-up interrupts"

Interrupt 98 is associated with JPEG Encoder on J722s. Remove this logic
from the shared AM62P and J722s device tree.

This reverts commit b8abee38c8362b6708e638dc01d8788f69afdf2b.

Fixes: b8abee38c836 ("TI: HACK: arm64: dts: ti: Configure am62p pinctrl-single for wake-up interrupts")
Reported-by: Brandon Brnich <b-brnich@ti.com>
Signed-off-by: Kendall Willis <k-willis@ti.com>
Tested-by: Brandon Brnich <b-brnich@ti.com>
---
 .../boot/dts/ti/k3-am62p-j722s-common-main.dtsi   | 15 +--------------
 1 file changed, 1 insertion(+), 14 deletions(-)

diff --git a/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi b/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi
index ec1ba80c755d9..95c928857934f 100644
--- a/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi
@@ -267,24 +267,11 @@
 	};
 
 	main_pmx0: pinctrl@f4000 {
-		compatible = "ti,am654-padconf", "pinctrl-single";
+		compatible = "pinctrl-single";
 		reg = <0x00 0xf4000 0x00 0x2b0>;
 		#pinctrl-cells = <1>;
 		pinctrl-single,register-width = <32>;
 		pinctrl-single,function-mask = <0xffffffff>;
-		/*
-		 * pinctrl IP DOES NOT give any functional IRQs when a
-		 * system is in active state. This IRQ is only a dummy IRQ
-		 * that is being used to trick Linux into thinking that
-		 * GIC can potentially recieve an interrupt from this IP.
-		 * This helps us setup the IO daisychain wakeups for deep
-		 * sleep via chained wake IRQs.
-		 * Please feel free to assign a different number here as
-		 * long as it is unused if 98 conflicts with another use case.
-		 */
-		interrupts = <GIC_SPI 98 IRQ_TYPE_LEVEL_HIGH>;
-		interrupt-controller;
-		#interrupt-cells = <1>;
 		bootph-all;
 	};
 
